rust/feature-flags/src/flag_definitions.rs

```diff
@@ -155,38 +151,55 @@ impl FeatureFlagList {
         client: Arc<dyn DatabaseClient + Send + Sync>,
         team_id: i32,
     ) -> Result<FeatureFlagList, FlagError> {
-        let mut conn = client.get_connection().await?;
-        // TODO: Clean up error handling here
+        let mut conn = client.get_connection().await.map_err(|e| {
+            tracing::error!("Failed to get database connection: {}", e);
+            FlagError::DatabaseUnavailable
+        })?;
 
         let query = "SELECT id, team_id, name, key, filters, deleted, active, ensure_experience_continuity FROM posthog_featureflag WHERE team_id = $1";
         let flags_row = sqlx::query_as::<_, FeatureFlagRow>(query)
             .bind(team_id)
             .fetch_all(&mut *conn)
-            .await?;
+            .await
+            .map_err(|e| {
+                tracing::error!("Failed to fetch feature flags from database: {}", e);
+                FlagError::Internal(format!("Database query error: {}", e))
+            })?;
 
-        let serialized_flags = serde_json::to_string(&flags_row).map_err(|e| {
-            tracing::error!("failed to serialize flags: {}", e);
-            println!("failed to serialize flags: {}", e);
-            FlagError::DataParsingError
-        })?;
+        let flags_list = flags_row
+            .into_iter()
+            .map(|row| {
+                let filters = serde_json::from_value(row.filters).map_err(|e| {
+                    tracing::error!("Failed to deserialize filters for flag {}: {}", row.key, e);
+                    FlagError::DataParsingError
+                })?;
 
-        let flags_list: Vec<FeatureFlag> =
-            serde_json::from_str(&serialized_flags).map_err(|e| {
-                tracing::error!("failed to parse data to flags list: {}", e);
-                println!("failed to parse data: {}", e);
+                Ok(FeatureFlag {
+                    id: row.id,
+                    team_id: row.team_id,
+                    name: row.name,
+                    key: row.key,
+                    filters,
+                    deleted: row.deleted,
+                    active: row.active,
+                    ensure_experience_continuity: row.ensure_experience_continuity,
```

Comment:
We might need the `rollout_percentage` to be serialized here.

Reply:
I'm not sure I totally follow; where in the feature flag row am I serializing the percentage?  That field is serialized here in the `FlagGroupType`
```
pub struct FlagGroupType {
    pub properties: Option<Vec<PropertyFilter>>,
    pub rollout_percentage: Option<f64>,
    pub variant: Option<String>,
}
```